specs/core/0_beacon-chain.md

```diff
@@ -1113,6 +1113,9 @@ def initialize_beacon_state_from_eth1(eth1_block_hash: Hash,
         eth1_data=Eth1Data(block_hash=eth1_block_hash, deposit_count=len(deposits)),
         latest_block_header=BeaconBlockHeader(body_root=hash_tree_root(BeaconBlockBody())),
     )
+    # Set the initial RANDAO mixes to hashes seeded by the Eth1 hash, to limit deposit ordering committee bias.
+    for i in range(MIN_SEED_LOOKAHEAD + 1):
+        state.randao_mixes[EPOCHS_PER_HISTORICAL_VECTOR - i - 1] = hash(eth1_block_hash + int_to_bytes(i, 8))
```

Comment:
What about

```python
    for i in range(EPOCHS_PER_HISTORICAL_VECTOR):
        state.randao_mixes[i] = eth1_block_hash
```

It seems to have equivalent security and it's more obvious what is going on (no `+ 1`, `- i - 1`, `int_to_bytes`, `hash`, or `MIN_SEED_LOOKAHEAD`).

Reply:
agreed here. no harm in backfilling the entire array. That is essentially what we had when they were all default value of ZERO

Reply:
ah, just saw that you are hashing in the eth1_block_hash with an integer index. I see some marginal value in doing that, but I think just filling as `eth1_block_hash` is fine.

Reply:
> I see some marginal value in doing that

Why? `get_seed ` already mixes in the `epoch` with `hash(domain_type + int_to_bytes(epoch, length=8) + mix)`.

Reply:
Ah missed that, thought there would be some assumption somewhere that every mix should be unique. Just copying over and completely filling is better then.